packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -504,9 +504,9 @@ class _HeroFlight {
           // supposed to end up then recreate the heroRect tween.
           final RenderBox finalRouteBox = manifest.toRoute.subtreeContext?.findRenderObject() as RenderBox;
           final Offset toHeroOrigin = toHeroBox.localToGlobal(Offset.zero, ancestor: finalRouteBox);
-          if (toHeroOrigin != heroRectTween.end.topLeft) {
-            final Rect heroRectEnd = toHeroOrigin & heroRectTween.end.size;
-            heroRectTween = _doCreateRectTween(heroRectTween.begin, heroRectEnd);
+          if (toHeroOrigin != heroRectTween.transform(1.0).topLeft) {
```

Comment:
For an animatable it's not guaranteed that 1.0 is the end and 0.0 is the beginning (see the docs).

Reply:
My initial goal was to be able to easily return chained tween from createRectTween callback.
I'm struggling to come up with better solution:
- Unfortunately there is no way to override `Animatable.chain` with generic `Tween.chain` with return type based on argument.
- Maybe it would be better to create `TweenChain` class to achieve this behavior (like `TweenSequence`)?

Reply:
Sorry, missed your first message. 
(comment for solution with generic `Tween.chain`)
just return a `Tween` is not an option since we should be able to pass `Animatable` and return type wouldn't be Tween then

another solution: add `chainTween` method to `Tween` with this behavior

Reply:
@goderbauer  Should I open another issue with clearly defined goal (use chained `Tween`s in` createRectTween`), but not the concrete implementation proposal, and move this discussion there?

Reply:
Yeah, maybe you can re-purpose the existing issue for that and change it to what the actual goal is.